### ISSUE TITLE
incorrect sorting in stitching_detailed.py

### ISSUE THREAD

**Issue by dave-andersen** [opencv/opencv#18650]
In samples/python/stitching_detailed.py, this code is supposed to grab the median focal value:
```
    focals = []
    for cam in cameras:
        focals.append(cam.focal)
    sorted(focals)
    if len(focals) % 2 == 1:
        warped_image_scale = focals[len(focals) // 2]
    else:
        warped_image_scale = (focals[len(focals) // 2] + focals[len(focals) // 2 - 1]) / 2
```

But it doesn't - sorted(focals) returns a copy of the array focals, but does not modify the original.

The code should instead read:
```
  focals.sort()
```

I assume it's better to keep this code as parallel as possible to the C++ version, but if someone wants to make a bigger change beyond fixing the bug, the initialization of focals here is non-pythonic, and would read better to me as,
```
  focals = sorted([cam.focal for cam in cameras])
```
which has the added benefit of making it harder to introduce the bug above.

(Or letting np do it, which exposes the idea it's looking for a median:

```warped_image_scale = float(np.median([c.focal for c in cameras]))```

Thanks as always for a fantastic framework!

**Comment by alalek** [comment]
Did you tried C++ version of this sample on your data?

/cc @@LaurentBerger

**Comment by dave-andersen** [comment]
Yup, the C++ version works well, and the code is correct.  It does an in-place sort of the focals:

```
    sort(focals.begin(), focals.end());
    float warped_image_scale;
    if (focals.size() % 2 == 1)
        warped_image_scale = static_cast<float>(focals[focals.size() / 2]);
    else
        warped_image_scale = static_cast<float>(focals[focals.size() / 2 - 1] + focals[focals.size() / 2]) * 0.5f;
```
(This sort(...) modifies focals, so the next part of the code correctly gets the median focal.)  The bug is just in the translation of the C++ to the Python version of the sample.

Tl;dr:  `sort(focals.begin(), focals.end())` in C++ modifies focals so that it is sorted;  `sorted(focals)` in Python does NOT modify focals;  the correct code to do the same as the C++ version is `focals.sort()`
